src/dataset/proof/repo/index.ts

```diff
@@ -0,0 +1,328 @@
+ *   (c) 2023 dataswap
+ *
+ *  Licensed under either the MIT License (the "MIT License") or the Apache License, Version 2.0
+ *  (the "Apache License"). You may not use this file except in compliance with one of these
+ *  licenses. You may obtain a copy of the MIT License at
+ *
+ *      https://opensource.org/licenses/MIT
+ *
+ *  Or the Apache License, Version 2.0 at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the MIT License or the Apache License for the specific language governing permissions and
+ *  limitations under the respective licenses.
+ ********************************************************************************/
+
+import {
+    DataType,
+    datasetProofEvm_Calibration,
+    datasetChallengeEvm_Calibration,
+    datasetProofEvm_Main,
+    datasetChallengeEvm_Main,
+} from "@dataswapjs/dataswapjs"
+import fs from "fs"
+import fastify from "fastify"
+
+import { handleEvmError, FileLock } from "../../../shared/utils/utils"
+import { DatasetProof, DatasetProofSubmitInfo } from "../types"
+
+/**
+ * Submits the dataset challenge proof to the blockchain network.
+ * @param network The network to submit the dataset proof to.
+ * @param datasetId The ID of the dataset.
+ * @param path The file path of the dataset proof.
+ * @returns A promise that resolves to true if the submission is successful, otherwise false.
+ */
+export async function submitDatasetChallengeProofs(
+    network: string,
+    datasetId: number,
+    path: string
+): Promise<boolean> {
+    const lock = new FileLock(String(datasetId) + path)
+    if (!lock.acquireLock()) {
+        console.log(
+            "Failed to acquire lock, another process may be using the file"
+        )
+    }
+
+    try {
+        console.log(
+            "Start submitDatasetChallengeProofs:",
+            "network:",
+            network,
+            "datasetId:",
+            datasetId,
+            "dataType:",
+            "path:",
+            path
+        )
+
+        const datasetChallengeProof = JSON.parse(
+            fs.readFileSync(path).toString()
+        )
+        const datasetChallengeEvm =
+            network === "calibration"
+                ? datasetChallengeEvm_Calibration
+                : datasetChallengeEvm_Main
+
+        const completed = await checkSubmissionChallengeProofsCriteria(
+            datasetChallengeEvm,
+            datasetId,
+            process.env.ADDRESS as string,
+            datasetChallengeProof.RandomSeed
+        )
+        if (completed) {
+            console.log("Dataset challenge proof had submited, do nothing~")
+            return true
+        }
+
+        datasetChallengeEvm.getWallet().add(process.env.PRIVATE_KEY as string)
+        await handleEvmError(
```

Comment:
Participation in the election and winning the election is required before submitting the challenge proof.